tests/utils/git_test.py

```diff
@@ -87,176 +163,75 @@ def test_git_version_too_old(self, run_subprocess_with_logging, run_subprocess):
         assert exc.value.args[0] == "Your git version is [1.0.0] but Rally requires at least git 1.9. Please update git."
         run_subprocess_with_logging.assert_called_with("git -C /src --version", level=logging.DEBUG)
 
-    @mock.patch("esrally.utils.io.ensure_dir")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_clone_successful(self, run_subprocess_with_logging, ensure_dir):
-        run_subprocess_with_logging.return_value = 0
-        src = "/src"
-        remote = "http://github.com/some/project"
-
-        git.clone(src, remote=remote)
-
-        ensure_dir.assert_called_with(src)
-        run_subprocess_with_logging.assert_called_with("git clone http://github.com/some/project /src")
-
-    @mock.patch("esrally.utils.io.ensure_dir")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_clone_with_error(self, run_subprocess_with_logging, ensure_dir):
-        run_subprocess_with_logging.return_value = 128
-        src = "/src"
-        remote = "http://github.com/some/project"
+    def test_clone_successful(self, teardown_clone):
+        git.clone(TestGit.tmp_clone_dir, remote=TestGit.remote_tmp_src_dir)
+        assert git.is_working_copy(TestGit.tmp_clone_dir)
 
+    def test_clone_with_error(self):
+        remote = "/this/remote/doesnt/exist.git"
         with pytest.raises(exceptions.SupplyError) as exc:
-            git.clone(src, remote=remote)
-        assert exc.value.args[0] == "Could not clone from [http://github.com/some/project] to [/src]"
-
-        ensure_dir.assert_called_with(src)
-        run_subprocess_with_logging.assert_called_with("git clone http://github.com/some/project /src")
+            git.clone(TestGit.tmp_clone_dir, remote=remote)
+        assert exc.value.args[0] == f"Could not clone from [{remote}] to [{TestGit.tmp_clone_dir}]"
 
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_fetch_successful(self, run_subprocess_with_logging):
-        run_subprocess_with_logging.return_value = 0
-        git.fetch("/src", remote="my-origin")
-        run_subprocess_with_logging.assert_called_with("git -C /src fetch --prune --tags my-origin")
+    def test_fetch_successful(self):
+        before_time = os.path.getmtime(os.path.join(TestGit.local_tmp_src_dir, ".git/FETCH_HEAD"))
+        git.fetch(TestGit.local_tmp_src_dir, remote=TestGit.remote_repo)
+        after_time = os.path.getmtime(os.path.join(TestGit.local_tmp_src_dir, ".git/FETCH_HEAD"))
+        fetch_head_modified = after_time > before_time
+        assert fetch_head_modified
```

Comment:
This seems brittle to me as it depends on the time resolution of the file system. I think it's good enough to notice that fetch does not fail.

Reply:
Sure. Addressed in https://github.com/elastic/rally/pull/1602/commits/c37f33409ec2f7668d01be524c4439f37480afdb